app/serializers/api/admin/customer_with_balance_serializer.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Api
+  module Admin
+    class CustomerWithBalanceSerializer < CustomerSerializer
+      attributes :balance, :balance_status
+
+      delegate :balance_value, to: :object
```

Comment:
This is just for my own understanding: We're delegating to the `:object` here, which is an instance of `Customer` - but the method `balance_value` isn't defined in the Customer class. So where does this delegation get `balance_value` from?

Reply:
This serializer relies on duck typing so `:object` must respond to `balance_value`, which is what the objects returned by `CustomersWithBalance#query` do. I should probably document this above line 8.

Reply:
Yeah I don't think it needs to be changed before merging, but it's a little hard to understand. Even knowing what you just said, and looking at `CustomersWithBalance`, the `query` method returns a scope on `Customer`, so I'm still not sure how we actually calculate the balance on a Customer; is it the `AS balance_value` line in `CustomersWithBalance` that's doing the heavy lifting?

I'm asking these questions knowing that I could probably search around and figure out exactly what's going on; but my point is that we could potentially make this part easier to understand without making someone do that (like the next person who comes along and looks at this, who will probably be one of us in six months 😁)

Reply:
Fair enough. I'll try to doc it in the class itself.

Reply:
Yeah, it's a bit tricky. The `balance_value` is in a SELECT statement and Rails makes all columns returned by the database available as methods.

Here is the sql for the select:
https://github.com/openfoodfoundation/openfoodnetwork/blob/99378e6407a285ed9bd2ec4ec47f2e057ff71f9b/app/services/customers_with_balance.rb#L23-L29

So that serializer can only be used when you used CustomersWithBalance to query the database with the balance_value.

Reply:
I'll work on this in a separate PR while this is being tested. I don't want this branch to live much longer and risk ending up with merge conflicts.

Reply:
Documented in https://github.com/openfoodfoundation/openfoodnetwork/pull/6642